### PR TITLE
Assessment.Sections.Metadata: Return full metadata

### PR DESCRIPTION
**Pull request by sorja**
# part of #1123 

## API
Returns tableSection with populated `tables: Table` with populated `rows: Array<Row>` and `rows` with populated `cols: Array<Col>`

`curl http://localhost:9000/api/countries/FIN/assessments/fra/cycles/2020/sections/extentOfForest/metadata`
```
{"id":"1","uuid":"cce9c7c9-51fe-428d-bc35-4592a3a4d5bc","props":{"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]},
"sectionId":"11","tables":[{"id":1,"rows":[{"id":1,"cols":[{"id":1,"uuid":"585d8d29-b69b-46d4-88e9-98f9ca63222d","props":{"index":0,"cycles":["b796f93e-054
f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","rowSpan":2,"labelKey":"extentOfForest.categoryHeader"},"r
owId":1},{"id":2,"uuid":"5d63f98f-7b6a-4f6b-bdc0-e4b41f2556b3","props":{"index":1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de
-8862320eb615"],"colSpan":null,"colType":"header","rowSpan":1,"labelKey":"extentOfForest.areaUnitLabel"},"rowId":1}],"uuid":"b6592e09-eaac-4b39-9ebe-0184be
cbc1d9","props":{"type":"header","index":"header_0","cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"linkToSection
":""},"tableId":1},{"id":2,"cols":[{"id":3,"uuid":"87e83877-5a3c-4641-ab0d-c6c643b1bff5","props":{"index":"header_0","cycles":["b796f93e-054f-483f-848e-b4b
392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"extentOfForest.forestArea"},"rowId":2},{"id":1279,"uuid":"402
bac09-a3a8-4c01-bd92-1b6490ec6ba9","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"1
990","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},{"id":1280,"uuid":"0a767f60-634d-4e4c-9724-d9ea95720252","props":{"index":1,"cycles":["b796f93
e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2000","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},{"id":1281,
"uuid":"9ceb8403-45bb-4e5f-a56c-adb20dc85f4c","props":{"index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"
colName":"2010","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},{"id":1282,"uuid":"52b93b54-ec15-4647-9def-7b2b74554c64","props":{"index":3,"cycles
":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2015","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},
{"id":1283,"uuid":"ee053db5-ca7e-4f25-b0fb-33a68b41e5dd","props":{"index":4,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-88623
20eb615"],"colName":"2016","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},{"id":1284,"uuid":"e509ffd6-d51c-434a-9bd5-ad0d70e7f460","props":{"index
":5,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2017","colSpan":1,"colType":"decimal","rowSpan":1},
"rowId":2},{"id":1285,"uuid":"351195da-8a30-4004-802b-bd48068fbafc","props":{"index":6,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6
-95de-8862320eb615"],"colName":"2018","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2},{"id":1286,"uuid":"f4dc2926-b575-4646-9bad-b59441119348","pro
ps":{"index":7,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2019","colSpan":1,"colType":"decimal","r
owSpan":1},"rowId":2},{"id":1287,"uuid":"a6f10b04-b77b-49a6-86dd-4fe6afae8e0c","props":{"index":8,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725
f-b3d4-45e6-95de-8862320eb615"],"colName":"2020","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":2}],"uuid":"dc574cf6-6724-435a-bfe4-600755775b53","pr
ops":{"type":"data","index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName":"forestArea","linkToS
ection":""},"tableId":1},{"id":3,"cols":[{"id":4,"uuid":"e06dea0c-fe4f-413d-908b-b09228c856f2","props":{"index":"header_0","cycles":["b796f93e-054f-483f-84
8e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"fraClass.otherWoodedLand"},"rowId":3},{"id":1288,"uuid"
:"02809957-9e58-40aa-8643-08a08ee8b038","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colNam
e":"1990","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3},{"id":1289,"uuid":"53ca5dac-7912-4f8e-908a-6708a29b6f64","props":{"index":1,"cycles":["b7
96f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2000","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3},{"id":
1290,"uuid":"4eaf464d-f6bd-4f9c-a823-7fc7a30e91c1","props":{"index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb61
5"],"colName":"2010","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3},{"id":1291,"uuid":"1d3fc60d-82b5-4211-a529-05ae0de484bf","props":{"index":3,"c
ycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2015","colSpan":1,"colType":"decimal","rowSpan":1},"rowId
":3},{"id":1292,"uuid":"041835cd-b157-4f50-a27e-0ccd33818ba5","props":{"index":4,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-
8862320eb615"],"colName":"2016","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3},{"id":1293,"uuid":"966e8706-e156-4ccc-9088-b02b9da79ae1","props":{"
index":5,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2017","colSpan":1,"colType":"decimal","rowSpan
":1},"rowId":3},{"id":1294,"uuid":"56c42834-be38-4d92-8aea-d2ee505ed10b","props":{"index":6,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4
-45e6-95de-8862320eb615"],"colName":"2018","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3},{"id":1295,"uuid":"85ac051f-2539-408a-827b-d83ad019c410"
,"props":{"index":7,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2019","colSpan":1,"colType":"decima
l","rowSpan":1},"rowId":3},{"id":1296,"uuid":"6ad03c9a-e6a3-4f82-83c9-b57c4b8947ef","props":{"index":8,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","46
82725f-b3d4-45e6-95de-8862320eb615"],"colName":"2020","colSpan":1,"colType":"decimal","rowSpan":1},"rowId":3}],"uuid":"f08b5d23-9c7a-454f-8deb-b4c2325fd4a3
","props":{"type":"data","index":1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName":"otherWoodedLand
","linkToSection":""},"tableId":1},{"id":4,"cols":[{"id":5,"uuid":"05783f95-c871-4f80-bcb7-2d76b23fedd2","props":{"index":"header_0","cycles":["b796f93e-05
4f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"fraClass.otherLand"},"rowId":4}],"uuid":"f20c
73a3-ccc8-4c53-a9fe-f0ddb77e167a","props":{"type":"data","index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]
,"variableName":"otherLand","linkToSection":""},"tableId":1},{"id":5,"cols":[{"id":6,"uuid":"0b571a24-b73c-4c7b-9fbf-e67b628754fb","props":{"index":"header
_0","cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"extentOfForest.tota
lLandArea"},"rowId":5}],"uuid":"d87888d9-ce48-4cbd-8452-5cbe6a40cfb4","props":{"type":"data","index":3,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","46
82725f-b3d4-45e6-95de-8862320eb615"],"variableName":"faoStat","linkToSection":""},"tableId":1},{"id":6,"cols":[{"id":7,"uuid":"ee0b551c-fcac-4a3d-9d60-c035
3a4da0dd","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"noticeMessage"
,"rowSpan":2,"labelKey":"extentOfForest.tableNoticeMessage"},"rowId":6}],"uuid":"4e352d4e-593a-4998-b335-b3fd0570201e","props":{"type":"noticeMessage","ind
ex":4,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"linkToSection":""},"tableId":1},{"id":241,"cols":[{"id":127
0,"uuid":"adc55f8a-a989-47c2-ae4a-e94585aedc89","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]
,"colName":"1990","colSpan":1,"colType":"header","rowSpan":1},"rowId":241},{"id":1271,"uuid":"1cee7651-1be2-40fa-a0c6-1828ffa19b62","props":{"index":1,"cyc
les":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2000","colSpan":1,"colType":"header","rowSpan":1},"rowId":2
41},{"id":1272,"uuid":"bfe280fb-e099-4c42-90ce-767df2acf165","props":{"index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8
862320eb615"],"colName":"2010","colSpan":1,"colType":"header","rowSpan":1},"rowId":241},{"id":1273,"uuid":"520e726a-c292-42b2-9429-4734ba7ee862","props":{"
index":3,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2015","colSpan":1,"colType":"header","rowSpan"
:1},"rowId":241},{"id":1274,"uuid":"a1886524-fdc2-43c2-acd6-26dcbcc90252","props":{"index":4,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d
4-45e6-95de-8862320eb615"],"colName":"2016","colSpan":1,"colType":"header","rowSpan":1},"rowId":241},{"id":1275,"uuid":"30bd91b2-b4a3-4991-bf38-01b4407e143
f","props":{"index":5,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2017","colSpan":1,"colType":"head
er","rowSpan":1},"rowId":241},{"id":1276,"uuid":"85cf3f63-a8e6-4faf-9a8f-8b9295b26815","props":{"index":6,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b",
"4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2018","colSpan":1,"colType":"header","rowSpan":1},"rowId":241},{"id":1277,"uuid":"d850a14f-c101-414d-a2c
6-107fcd6e3ab3","props":{"index":7,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2019","colSpan":1,"c
olType":"header","rowSpan":1},"rowId":241},{"id":1278,"uuid":"baf7442e-188d-4614-b2ee-83c856015a9b","props":{"index":8,"cycles":["b796f93e-054f-483f-848e-b
4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"2020","colSpan":1,"colType":"header","rowSpan":1},"rowId":241}],"uuid":"b246ba5c-55d7-4955-
9e2d-2acc246c81fc","props":{"type":"header","index":"header_1","cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]},"t
ableId":1}],"uuid":"cd8f196d-a542-46c5-8683-b1e77acebada","props":{"odp":true,"name":"extentOfForest","unit":"haThousand","cycles":["b796f93e-054f-483f-848
e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]},"tableSectionId":1},{"id":2,"rows":[{"id":8,"cols":[{"id":8,"uuid":"d7ff3ed7-d414-45bb-b5f7-e4cb94
d3cea3","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","rowSpan
":1,"labelKey":"climaticDomain.climaticDomain"},"rowId":8},{"id":9,"uuid":"72a334c9-9c3a-4466-956c-b91d20c3399f","props":{"index":1,"cycles":["b796f93e-054
f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","rowSpan":1,"labelKey":"climaticDomain.percentOfForestArea
2015"},"rowId":8},{"id":10,"uuid":"c235a8cb-4854-44d5-8c8f-26de97a64b1e","props":{"index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4
-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","rowSpan":1,"labelKey":"climaticDomain.percentOfForestArea2015Override"},"rowId":8}],"uuid":"774cd
c0a-aa9d-41c5-99cd-b4c47d525a68","props":{"type":"header","index":"header_0","cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862
320eb615"],"linkToSection":""},"tableId":2},{"id":9,"cols":[{"id":11,"uuid":"96bba052-c841-4b06-8d97-946951e22bd6","props":{"index":"header_0","cycles":["b
796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"climaticDomain.boreal"},"rowId":9},
{"id":12,"uuid":"0d6db48b-e0c3-438c-abb8-94926245f72c","props":{"index":-1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-886232
0eb615"],"colType":"calculated"},"rowId":9},{"id":13,"uuid":"520236d5-7da2-455d-abe0-dfe37c96c4a1","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4
b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"percentOfForestArea2015","colType":"decimal"},"rowId":9}],"uuid":"ad2e04ba-fd96-4c16-a3d3-f3
8900809444","props":{"type":"data","index":0,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName":"borea
l","linkToSection":""},"tableId":2},{"id":10,"cols":[{"id":14,"uuid":"689e6609-e0c5-491c-876b-8da60180cd90","props":{"index":"header_0","cycles":["b796f93e
-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"climaticDomain.temperate"},"rowId":10},{"i
d":15,"uuid":"4113132b-9638-4fe8-8df8-5e6439a7d6cb","props":{"index":-1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb
615"],"colType":"calculated"},"rowId":10},{"id":16,"uuid":"97ad832d-ec89-495d-af5d-b2706611a1f0","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-b4b3
92c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"percentOfForestArea2015","colType":"decimal"},"rowId":10}],"uuid":"9ea48a4a-1ce6-4bb2-9540-110
c8fee180f","props":{"type":"data","index":1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName":"temper
ate","linkToSection":""},"tableId":2},{"id":11,"cols":[{"id":17,"uuid":"d3930e02-8137-4ec5-a966-91e74da0ebe6","props":{"index":"header_0","cycles":["b796f9
3e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"climaticDomain.subtropical"},"rowId":11}
,{"id":18,"uuid":"2ea86e0a-c8b4-4d47-835e-e57ff22a390e","props":{"index":-1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-88623
20eb615"],"colType":"calculated"},"rowId":11},{"id":19,"uuid":"768e23f9-a1ef-4b55-9c03-ee95773e71e5","props":{"index":0,"cycles":["b796f93e-054f-483f-848e-
b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"percentOfForestArea2015","colType":"decimal"},"rowId":11}],"uuid":"a4d31236-812a-4b91-98e9
-d9ff1eae035d","props":{"type":"data","index":2,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName":"su
b_tropical","linkToSection":""},"tableId":2},{"id":12,"cols":[{"id":20,"uuid":"a6f7e4bb-4cb0-4b45-a2fb-a92825209bc6","props":{"index":"header_0","cycles":[
"b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colSpan":1,"colType":"header","labelKey":"climaticDomain.tropical"},"rowId"
:12},{"id":21,"uuid":"7c2d22cf-d2cc-4a33-8fc5-17afafdef2b9","props":{"index":-1,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8
862320eb615"],"colType":"calculated"},"rowId":12},{"id":22,"uuid":"71f45a1f-94c6-4fd4-9bf3-ea8bb69ac703","props":{"index":0,"cycles":["b796f93e-054f-483f-8
48e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"colName":"percentOfForestArea2015","colType":"decimal"},"rowId":12}],"uuid":"f1f674e7-606d-4fe1-
a10f-c9df0e1cad72","props":{"type":"data","index":3,"cycles":["b796f93e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"],"variableName"
:"tropical","linkToSection":""},"tableId":2}],"uuid":"b556133a-1c3b-4b41-818b-7754d8e5804a","props":{"odp":false,"name":"climaticDomain","cycles":["b796f93
e-054f-483f-848e-b4b392c7ab2b","4682725f-b3d4-45e6-95de-8862320eb615"]},"tableSectionId":1}]}
```

### CODE REVIEW

**Review by minotogna — CHANGES_REQUESTED** [review]
just use one endpoint to get section metadata. getSectionMetadata endpoint must return tableSections, tables, rows, cols.
Add all 4 to the response in one PR

File: server/controller/assessment/getTablesMetadata.ts [hunk 1]
```diff
@@ -1,6 +1,44 @@
...
+ * row.tableId = table.id
+ * @param table
+ * @param assessment
+ */
+const _populateWithRows = async (table: Table, assessment: Assessment): Promise<Table> => ({
+  ...table,
+  rows: await _getManyRows({ assessment, tableId: table.id }),
+})
 
 export const getTablesMetadata = async (
```
Comment (minotogna) [inline server/controller/assessment/getTablesMetadata.ts:43]: call this method getSectionMetadata and use just this query:

```sql
with ts as (
    select ts.*
    from assessment_fra.table_section ts
             left join assessment_fra.section s on ts.section_id = s.id
    where s.props ->> 'name' = 'extentOfForest' 
),
     t as (
         select t.*
         from assessment_fra."table" t,
              ts
         where t.table_section_id = ts.id
     ),
     rows as (
         select r.*, jsonb_agg(c.* order by c.id) as cols
         from assessment_fra.col c
                  left join assessment_fra.row r on r.id = c.row_id
         where r.table_id in (select t.id from t)
         group by r.id
     ),
     tables as (
         select t.*, jsonb_agg(r.* order by r.id) as rows
         from "rows" r
                  left join t on t.id = r.table_id
         group by t.id, t.uuid, t.props, t.table_section_id
     )
select ts.*,
       jsonb_agg(t.* order by t.id) as tables
from "tables" t
         left join ts on t.table_section_id = ts.id
group by ts.id, ts.uuid, ts.props, ts.section_id
```

**Review by minotogna — CHANGES_REQUESTED** [review]

**Comment by sorja** [comment]
Draft: the return from query doesn't work out of the box for some reason, changing $1 to 'extentOfForest' returns correct result, but changing it to dynamic value from values (= `$1` ) and passing the same sectionName ('extentOfForest') quits silently

Investigating and opening from draft once solved

File: server/repository/assessment/getSectionMetaData.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  props: {
+    assessment: Assessment
+    section: string
+  },
+  client: BaseProtocol = DB
+): Promise<TableSection> => {
+  const { section, assessment } = props
+  const schemaName = Schemas.getName(assessment)
+
+  return client.oneOrNone<TableSection>(
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:15]: this returns an array of TableSection

File: server/repository/assessment/getSectionMetaData.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    `
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:16]: forgot to add the filter by cycle. add it to all sub queries

**Comment by minotogna** [comment]
> Draft: the return from query doesn't work out of the box for some reason, changing $1 to 'extentOfForest' returns correct result, but changing it to dynamic value from values (= `$1` ) and passing the same sectionName ('extentOfForest') quits silently
> 
> Investigating and opening from draft once solved

and then use 
```sql
where s.props ->> 'name' = '${tableName}'
```
 just to make it work quickly ?

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/repository/assessment/index.ts [hunk 4]
```diff
@@ -18,7 +20,9 @@ export const AssessmentRepository = {
   createAssessmentSchema,
   read,
   readSections,
-  readTablesMetadata,
+  getSectionMetaData,
+  getManyRows,
+  getManyCols,
```
Comment (minotogna) [inline server/repository/assessment/index.ts:25]: do we still need these?

**Review by minotogna — CHANGES_REQUESTED** [review]
address my comments, or leave a message if they do not apply

File: server/repository/assessment/getSectionMetaData.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            from ${schemaName}.table_section ts
+                     left join ${schemaName}.section s on ts.section_id = s.id
+            where s.props ->> 'name' = $1
+              and s.props -> 'cycles' ? $2
+        ),
+             t as (
+                 select t.*
+                 from ${schemaName}."table" t,
+                      ts
+                 where t.table_section_id = ts.id and ts.props -> 'cycles' ? $2
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:29]: t.props -> 'cycles' ? $2

File: server/repository/assessment/getSectionMetaData.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  const schemaName = Schemas.getName(assessment)
+
+  return client.oneOrNone<Array<TableSection>>(
+    `
+        with ts as (
+            select ts.*
+            from ${schemaName}.table_section ts
+                     left join ${schemaName}.section s on ts.section_id = s.id
+            where s.props ->> 'name' = $1
+              and s.props -> 'cycles' ? $2
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:23]: ts.props -> 'cycles' ? $2

File: server/repository/assessment/getSectionMetaData.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+                 select t.*
+                 from ${schemaName}."table" t,
+                      ts
+                 where t.table_section_id = ts.id and ts.props -> 'cycles' ? $2
+             ),
+             rows as (
+                 select r.*, jsonb_agg(c.* order by c.id) as cols
+                 from ${schemaName}.col c
+                          left join ${schemaName}.row r on r.id = c.row_id
+                 where r.table_id in (select t.id from t) and r.props -> 'cycles' ? $2
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:35]: r.props -> 'cycles' ? $2 and c.props -> 'cycles' ? $2

File: server/repository/assessment/getSectionMetaData.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                 group by r.id
+             ),
+             tables as (
+                 select t.*, jsonb_agg(r.* order by r.id) as rows
+                 from "rows" r
+                          left join t on t.id = r.table_id
+                 where t.props -> 'cycles' ? $2
```
Comment (minotogna) [inline server/repository/assessment/getSectionMetaData.ts:42]: here no need

**Review by minotogna — CHANGES_REQUESTED** [review]

File: server/repository/assessment/getSectionMetaData.ts (hunk 7 above)
Comment (sorja) [reply server/repository/assessment/getSectionMetaData.ts:35]: Adding `c.props -> cycles` filter for some breaks the query and returns nothing

Comment (minotogna) [reply server/repository/assessment/getSectionMetaData.ts]: why did you commit it it then?
have you tried to add the filter inside the jsonb_agg(c.* order by c.id)

Comment (sorja) [reply server/repository/assessment/getSectionMetaData.ts]: it works, i resolved some issue after commenting (resolved comment)

**Review by minotogna — APPROVED** [review]

**Pull request merged**